From c2e580f1f3137d8517894ba6d5cd6d7f85682587 Mon Sep 17 00:00:00 2001
Message-Id: <c2e580f1f3137d8517894ba6d5cd6d7f85682587.1387298827.git.minovotn@redhat.com>
In-Reply-To: <3ed0fb61a3dc912ef036d7ef450bed192090709e.1387298827.git.minovotn@redhat.com>
References: <3ed0fb61a3dc912ef036d7ef450bed192090709e.1387298827.git.minovotn@redhat.com>
From: "Michael S. Tsirkin" <mst@redhat.com>
Date: Tue, 17 Dec 2013 15:17:13 +0100
Subject: [PATCH 10/56] fw_cfg: move typedef to qemu/typedefs.h

RH-Author: Michael S. Tsirkin <mst@redhat.com>
Message-id: <1387293161-4085-11-git-send-email-mst@redhat.com>
Patchwork-id: 56315
O-Subject: [PATCH qemu-kvm RHEL7.0 v2 10/57] fw_cfg: move typedef to qemu/typedefs.h
Bugzilla: 1034876
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>
RH-Acked-by: Marcel Apfelbaum <marcel.a@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

Less header dependencies this way.

Signed-off-by: Michael S. Tsirkin <mst@redhat.com>
(cherry picked from commit 45936c8b79012da4c8986c20af4afcc9cd14bb8e)
---
 include/hw/i386/pc.h      | 1 -
 include/hw/nvram/fw_cfg.h | 2 +-
 include/qemu/typedefs.h   | 1 +
 3 files changed, 2 insertions(+), 2 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 include/hw/i386/pc.h      | 1 -
 include/hw/nvram/fw_cfg.h | 2 +-
 include/qemu/typedefs.h   | 1 +
 3 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index 4175c1b..2cf7baa 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -9,7 +9,6 @@
 #include "net/net.h"
 #include "exec/memory.h"
 #include "hw/i386/ioapic.h"
-#include "hw/nvram/fw_cfg.h"
 
 /* PC-style peripherals (also used by other machines).  */
 
diff --git a/include/hw/nvram/fw_cfg.h b/include/hw/nvram/fw_cfg.h
index 48a8aa4..f60dd67 100644
--- a/include/hw/nvram/fw_cfg.h
+++ b/include/hw/nvram/fw_cfg.h
@@ -6,6 +6,7 @@
 #include <stddef.h>
 
 #include "exec/hwaddr.h"
+#include "qemu/typedefs.h"
 #endif
 
 #define FW_CFG_SIGNATURE        0x00
@@ -60,7 +61,6 @@ typedef struct FWCfgFiles {
 
 typedef void (*FWCfgCallback)(void *opaque, uint8_t *data);
 
-typedef struct FWCfgState FWCfgState;
 void fw_cfg_add_bytes(FWCfgState *s, uint16_t key, void *data, size_t len);
 void fw_cfg_add_string(FWCfgState *s, uint16_t key, const char *value);
 void fw_cfg_add_i16(FWCfgState *s, uint16_t key, uint16_t value);
diff --git a/include/qemu/typedefs.h b/include/qemu/typedefs.h
index 2e7d942..a332d88 100644
--- a/include/qemu/typedefs.h
+++ b/include/qemu/typedefs.h
@@ -63,6 +63,7 @@ typedef struct EventNotifier EventNotifier;
 typedef struct VirtIODevice VirtIODevice;
 typedef struct QEMUSGList QEMUSGList;
 typedef struct SHPCDevice SHPCDevice;
+typedef struct FWCfgState FWCfgState;
 typedef struct Range Range;
 
 #endif /* QEMU_TYPEDEFS_H */
-- 
1.7.11.7

